ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -500,6 +500,10 @@ public static Hive getThreadLocal() {
     return hiveDB.get();
   }
 
+  public static Hive getCurrHiveDb(Hive hiveDb, HiveConf c) throws HiveException {
```

Comment:
Do we really want to have a new method for this?

What if we modify the `get`, or the `getThreadLocal` to provide an always open `Hive`?